### DISCUSSION TITLE
New codespace link for Jupyter lab?

### DISCUSSION THREAD

**Discussion by pamelafox** [community/community#57038] [Codespaces] [answered]
### Select Topic Area

Question

### Body

We are very excited about the "Open in JupyterLab" option for Codespaces. We have a single repo for all class assignments (notebook files) and would like to be able to link from the syllabus directly to JupyterLab for each assignment. Thus, we're wondering if:

1) Is it possible to create a codespaces/new link that always opens in Jupyterlab? I tried ?editor=jupyter but that didn't seem to get forwarded on once the Codespace was being built.
2) Is it possible to specify the filepath that gets opened once Jupyterlab is open? That way, students will be immediately looking at the relevant file and not have to dig around the folders. I know it's possible to specify an open file for VS Code in devcontainer.json, but we're hoping to customize it in the link alone, so that we can have the same repo but a different link for each assignment.

Thanks!

**Comment by alexagillman** [comment, marked as answer]
Hi, we currently don't support deep linking to a specific editor, but we are hoping to add this functionality at some point in the future

**Reply by pamelafox** [reply]
Thanks @alexagillman ! Just want to double check since I'm meeting with many Jupyterlab-interested teachers this week: the only ways to open a repo in Jupyterlab are:

1) *first* create the Codespace (which opens in VS Code in web), and then find the created codespace in your codespaces list and re-open it in Jupyterlab.
2) configure GitHub account to have Jupyterlab as default editor (which might make sense for someone who primarily works with notebooks).

Thanks!

**Reply by alexagillman** [reply]
Yes that is correct!

**Reply by palo-manel** [reply]
This would be a very welcome feature! It would enable a [Binder](https://mybinder.org/)-like experience for our secure GitHub Enterprise organization.

**Reply by paololazzari** [reply]
I think it would be better if it was up to the repository to configure whether the codespace should be opened in VSCode or Jupyterlab by default, as opposed to this being dependent on the users preference.

**Comment by jsulopzs** [comment]
I'm facing the same issue as @pamelafox here.

I tried to create a custom `.devcontainer.json` so that the opened code editor is `Jupyter Lab`, but I haven't had any success.

**Comment by palo-manel** [comment]
I agree with @pamelafox the most obvious way to expose this functionality would be enabling `?editor=jupyter` for the "Open in JupyterLab" deep link.
